src/ert/sensitivity_analysis/design_matrix.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import pandas as pd
+import xarray as xr
+
+from ert.config.gen_kw_config import GenKwConfig, TransformFunctionDefinition
+
+if TYPE_CHECKING:
+    from ert.config import (
+        ErtConfig,
+    )
+    from ert.storage import LocalEnsemble, LocalStorage
+
+DESIGN_MATRIX_GROUP = "DESIGN_MATRIX"
+
+
+def read_design_matrix(
+    ert_config: ErtConfig,
+    xlsfilename: Path | str,
+    designsheetname: str = "DesignSheet01",
+    defaultssheetname: str = "DefaultValues",
+) -> pd.DataFrame:
+    """
+    Reads out all file content from different files and create dataframes
+    """
+    design_matrix_sheet = _read_excel(xlsfilename, designsheetname)
+    if "REAL" in design_matrix_sheet.columns:
+        design_matrix_sheet.set_index(design_matrix_sheet["REAL"])
+        del design_matrix_sheet["REAL"]
+    try:
+        _validate_design_matrix_header(design_matrix_sheet)
+    except ValueError as err:
+        raise ValueError(f"Design matrix not valid, error: {err!s}") from err
+
+    # Todo: Check for invalid realizations, drop them maybe?
+
+    if designsheetname == defaultssheetname:
+        raise ValueError("Design-sheet and defaults-sheet can not be the same")
+
+    # This should probably handle/(fill in) missing values in design_matrix_sheet as well
```

Comment:
When merging suggest adding this as an issue instead (which is fine, dont have to fix it in this PR). Same goes for the other comments